src/script_error.rs

```diff
@@ -30,23 +33,40 @@ pub enum ScriptError {
     OpReturn,
 
     // Max sizes
-    #[error("Script size exceeded maximum")]
-    ScriptSize,
+    #[error(
+        "Script size{} exceeded maxmimum ({} bytes)",
+        .0.map_or("", |size| " ({size} bytes)"),
+        script::MAX_SCRIPT_SIZE
+    )]
+    ScriptSize(Option<usize>),
 
-    #[error("push size exceeded maximum")]
-    PushSize,
+    #[error(
+        "push size{} exceeded maxmimum ({} bytes)",
+        .0.map_or("", |size| " ({size} bytes)"),
+        script::MAX_SCRIPT_ELEMENT_SIZE
+    )]
+    PushSize(Option<usize>),
 
-    #[error("operation count exceeded maximum")]
+    /// __NB__: This doesn’t take an “actual count” argument, because `OpCount` depends on
+    ///         conditional execution and thus can only be checked incrementally. However, we could
+    ///         statically check a “minimum operation count” for a script, which could then include
+    ///         the minimum in the error.
+    #[error("operation count exceeded maxmimum of {}", interpreter::MAX_OP_COUNT)]
     OpCount,
 
-    #[error("stack size exceeded maximum")]
-    StackSize,
+    #[error(
+        "stack depth exceeded maxmimum of {} entries",
+        interpreter::MAX_STACK_DEPTH
+    )]
+    StackSize(Option<TryFromIntError>),
 
-    #[error("signature count exceeded maximum")]
-    SigCount,
+    #[error("signature count wasn’t in the range [1, {}]{}",
```

Comment:
This is clearer in the context of Rust:
```suggestion
    #[error("signature count wasn’t in the range 1..={}{}",
```